ui/pages/setup_secure.tsx

```diff
@@ -48,19 +48,15 @@ const SetupSecure: NextPage = () => {
           ? returnTo.toString()
           : window.location.pathname.replace("setup_secure", "setup_complete"),
       })
-      .then(async ({ data }) => {
-        const pwParam = pwChanged ? pwChanged.toString() : "false";
-        router.query.flow = data.id;
-        router.query.pw_changed = pwParam;
-
-        await router.replace(
-          {
-            pathname: window.location.pathname,
-            query: router.query,
-          },
-          undefined,
-          { shallow: true },
-        );
+      .then(({ data }) => {
+        const pwParam = pwChanged ? `&pw_changed=${pwChanged.toString()}` : "";
+        if (flowId !== data.id) {
+          window.history.replaceState(
```

Comment:
question: Is `router.replace` and `window.history.replaceState` the same API and the 1st is the nextjs way of interacting with the history API? In that case why don't we use the tool provided from the framework?

Reply:
Basically: yes. The problem with next router is we can't use relative URLs. I don't see any advantage over using the router that using the vanilla api.

Reply:
Cool, thanks